### PR TITLE
Text selection UI matches behavior on Android.

### PR DESCRIPTION
**Pull request by mpcomplete**
- Handles appear with tap or long press.
- Toolbar appears with long press on text, or tap on handle.
- Correct toolbar items shown depending on context.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/editable_line.dart [hunk 1]
```diff
@@ -116,6 +116,7 @@ class RenderEditableLine extends RenderBox {
   set paintOffset(Offset value) {
     if (_paintOffset == value)
       return;
+    print('paintOffset changed');
```
Comment (eseidelGoogle) [inline packages/flutter/lib/src/rendering/editable_line.dart]: Did you mean to leave this in?

Comment (mpcomplete) [inline packages/flutter/lib/src/rendering/editable_line.dart]: Nope!

**Comment by Hixie** [comment]
I'm not super familiar with the underlying code, but LGTM, modulo travis anger.

**Comment by eseidelGoogle** [comment]
This now contains commits you didn't intend.  I think you want to rebase.  possibly `git rebase -i` to re-order, squash, etc.

**Pull request merged**